src/ngAnimate/animateQueue.js

```diff
@@ -624,25 +638,29 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
         // there is no need to continue traversing at this point
         if (parentAnimationDetected && animateChildren === false) break;
 
-        if (!rootElementDetected) {
-          // angular doesn't want to attempt to animate elements outside of the application
-          // therefore we need to ensure that the rootElement is an ancestor of the current element
-          rootElementDetected = isMatchingElement(parentElement, $rootElement);
-          if (!rootElementDetected) {
-            parentHost = parentElement.data(NG_ANIMATE_PIN_DATA);
-            if (parentHost) {
-              parentElement = parentHost;
-              rootElementDetected = true;
-            }
+        if (!rootElementDetected && !parentHost) {
```

Comment:
Good catch. I think we can add this as a reasonable breaking change. So far, pin() was broken for more cases than it was working for, so adoption doesn't seem to high.